src/components/AddProductStoreToGroupModal.vue

```diff
@@ -0,0 +1,186 @@
+          "noConditionFind": "Y",
+          "filterByDate": 'Y'
+        })
+
+        if(!hasError(resp)) {
+          this.selectedProductStores = resp.data.docs
+          this.selectedProductStoreValues = JSON.parse(JSON.stringify(resp.data.docs))
+        } else {
+          throw resp.data
+        }
+      } catch(err) {
+        logger.error(err)
+      }
+    },
+    isSelected(productStoreId: any) {
+      return this.selectedProductStoreValues.some((productStore: any) => productStore.productStoreId === productStoreId);
+    },
+    toggleProductStoreSelection(store: any) {
+      let selectedStore = this.selectedProductStoreValues.some((productStore: any) => productStore.productStoreId === store.productStoreId);
+
+      if(selectedStore) {
+        this.selectedProductStoreValues = this.selectedProductStoreValues.filter((productStore: any) => productStore.productStoreId !== store.productStoreId);
+      } else {
+        this.selectedProductStoreValues.push(store);
+      }
+    },
+    async saveProductStores() {
+      const productStoresToAdd = this.selectedProductStoreValues.filter((selectedStore: any) => !this.selectedProductStores.some((store: any) => store.productStoreId === selectedStore.productStoreId))
+      const productStoresToRemove = this.selectedProductStores.filter((store: any) => !this.selectedProductStoreValues.some((selectedStore: any) => store.productStoreId === selectedStore.productStoreId))
+
+      const removeResponses = await Promise.allSettled(productStoresToRemove
+        .map(async (store: any) => await FacilityService.updateProductStoreFacilityGroup({
+          "productStoreId": store.productStoreId,
+          "facilityGroupId": this.group.facilityGroupId,
+          "fromDate": store.fromDate,
+          "thruDate": DateTime.now().toMillis()
+        }))
+      )
+
+      const addResponses = await Promise.allSettled(productStoresToAdd
+        .map(async (store: any) => await FacilityService.createProductStoreFacilityGroup({
+          "productStoreId": store.productStoreId,
+          "facilityGroupId": this.group.facilityGroupId,
+          "fromDate": DateTime.now().toMillis()
+        }))
+      )
+
+      const hasFailedResponse = [...removeResponses, ...addResponses].some((response: any) => response.status === 'rejected')
+      if (hasFailedResponse) {
+        showToast(translate("Failed to associate some product stores to group."))
+      } else {
+        showToast(translate("Product stores associated to group successfully."))
+      }
+      this.fetchGroupsCount()
+      modalController.dismiss()
+    },
+    async fetchGroupsCount() {
+      const productStoreCountByGroup = await FacilityService.fetchProductStoreCountByGroup([this.group.facilityGroupId])
+      this.groups.map((group: any) => {
+        if(group.facilityGroupId === this.group.facilityGroupId) {
+          group.productStoreCount = productStoreCountByGroup[this.group.facilityGroupId]
+        }
+      })
```

Comment:
Use find to update the group information, as when using map the iteration will not stop even when we found the facilityGroup and the iteration will go on for all the items in the groups array.